wemix/bind/structs.go

```diff
@@ -0,0 +1,612 @@
+}
+
+func (gov *GovContracts) Address() struct {
+	Registry      common.Address
+	Gov           common.Address
+	Staking       common.Address
+	BallotStorage common.Address
+	EnvStorage    common.Address
+} {
+	return gov.address
+}
+
+func (src *GovContracts) Copy(dst *GovContracts, backend bind.ContractBackend) error {
+	if dst == nil {
+		return errors.New("nil destination")
+	}
+	zeroAddress := common.Address{}
+	errZeroAddress := errors.New("invalid src address")
+
+	srcAddress := src.Address()
+	if address := srcAddress.Registry; address == zeroAddress {
+		return errors.Wrap(errZeroAddress, "Registry")
+	} else if contract, err := NewRegistry(address, backend); err != nil {
+		return errors.Wrap(err, "Registry")
+	} else {
+		dst.Registry, dst.address.Registry = contract, address
+	}
+
+	if address := srcAddress.Gov; address == zeroAddress {
+		return errors.Wrap(errZeroAddress, "Gov")
+	} else if proxy, err := NewGov(address, backend); err != nil {
+		return errors.Wrap(err, "Gov")
+	} else if imp, err := NewGovImp(address, backend); err != nil {
+		return errors.Wrap(err, "GovImp")
+	} else {
+		dst.Gov, dst.GovImp, dst.address.Gov = proxy, imp, address
+	}
+
+	if address := srcAddress.Staking; address == zeroAddress {
+		return errors.Wrap(errZeroAddress, "Staking")
+	} else if proxy, err := NewStaking(address, backend); err != nil {
+		return errors.Wrap(err, "Staking")
+	} else if imp, err := NewStakingImp(address, backend); err != nil {
+		return errors.Wrap(err, "StakingImp")
+	} else {
+		dst.Staking, dst.StakingImp, dst.address.Staking = proxy, imp, address
+	}
+
+	if address := srcAddress.BallotStorage; address == zeroAddress {
+		return errors.Wrap(errZeroAddress, "BallotStorage")
+	} else if proxy, err := NewBallotStorage(address, backend); err != nil {
+		return errors.Wrap(err, "BallotStorage")
+	} else if imp, err := NewBallotStorageImp(address, backend); err != nil {
+		return errors.Wrap(err, "BallotStorageImp")
+	} else {
+		dst.BallotStorage, dst.BallotStorageImp, dst.address.BallotStorage = proxy, imp, address
+	}
+
+	if address := srcAddress.EnvStorage; address == zeroAddress {
+		return errors.Wrap(errZeroAddress, "EnvStorage")
+	} else if proxy, err := NewEnvStorage(address, backend); err != nil {
+		return errors.Wrap(err, "EnvStorage")
+	} else if imp, err := NewEnvStorageImp(address, backend); err != nil {
+		return errors.Wrap(err, "EnvStorageImp")
+	} else {
+		dst.EnvStorage, dst.EnvStorageImp, dst.address.EnvStorage = proxy, imp, address
+	}
+
+	return nil
+}
+
+func (src *GovContracts) Equal(dst *GovContracts) bool {
+	return bytes.Equal(src.address.Registry.Bytes(), dst.address.Registry.Bytes()) &&
+		bytes.Equal(src.address.Gov.Bytes(), dst.address.Gov.Bytes()) &&
+		bytes.Equal(src.address.Staking.Bytes(), dst.address.Staking.Bytes()) &&
+		bytes.Equal(src.address.BallotStorage.Bytes(), dst.address.BallotStorage.Bytes()) &&
+		bytes.Equal(src.address.EnvStorage.Bytes(), dst.address.EnvStorage.Bytes())
+}
```

Comment:
To compare all contracts' addresses without missing any, it is recommended to do the following

```suggestion
func (src *GovContracts) Equal(dst *GovContracts) bool {
	return reflect.DeepEqual(src.address,  dst.address)
}
```